### DISCUSSION TITLE
Missing preview images for images larger than 600 KB on my Mastodon server

### DISCUSSION THREAD

**Discussion by wonderfulpayments** [mastodon/mastodon#25354] [General]
I noticed we are missing preview images for some posts on my Mastodon server. I have checked the meta tags, and they are correctly set. I finally narrowed the issue down to the size of the images on the URls we post on our Mastodon server. I can see that URLs with images (mostly png) larger than 600 KB (my estimate) are not getting displayed in the preview pane. If I convert the PNG images to JPEG format and then reupload them, the link preview image is displayed properly. 

Has anyone else noticed this and any idea how to resolve this?

![Screenshot 2023-06-09 at 16 48 40](https://github.com/mastodon/mastodon/assets/71023177/307439c6-5f4b-43ec-a750-e729ef363e31)

**Comment by rgsteele** [comment]
What are you using for object storage? If it's an S3-compatible backend, you may need to set the `S3_FORCE_SINGLE_REQUEST=true` environment variable documented at https://docs.joinmastodon.org/admin/optional/object-storage/#S3.
